src/app/convai-checker.component.spec.ts

```diff
@@ -1455,4 +1500,394 @@ describe('Convai checker test', () => {
     // Send an input event to trigger the service call.
     setTextAndFireInputEvent(queryTexts[callCount], textArea);
   }));
+
+  it('Test circle square diamond change for score thresholds', async(() => {
+    let fixture = TestBed.createComponent(ConvaiCheckerCustomDemoSettingsTestComponent);
+
+    // Configure settings.
+    let demoSettings = getCopyOfDefaultDemoSettings();
+    demoSettings.scoreThresholds = [0, 0.6, 0.8];
+    demoSettings.loadingIconStyle = LoadingIconStyle.CIRCLE_SQUARE_DIAMOND;
+    fixture.componentInstance.setDemoSettings(demoSettings);
+
+    fixture.detectChanges();
+    let checker = fixture.componentInstance.checker;
+
+    // Set up the mock responses for the series of three requests that will be
+    // made in the test.
+    let queryTexts = [
+      'Your mother was a hamster',
+      'Your father smelled of elderberries',
+      'What is the air velocity of an unladen swallow?'
+    ];
+
+    let callCount = 0;
+    let mockResponses = [
+      getMockCheckerResponseWithScore(0.9, checker.getToken(queryTexts[0])),
+      getMockCheckerResponseWithScore(0.7, checker.getToken(queryTexts[1])),
+      getMockCheckerResponseWithScore(0.2, checker.getToken(queryTexts[2]))
+    ];
+
+    let mockBackend = TestBed.get(MockBackend);
+    mockBackend.connections
+     .subscribe((connection: MockConnection) => {
```

Comment:
Something about this implicit loop is still bothering me a tad, and I'm  not sure if there's actually a good way to improve it.  If they don't need to be called in this exact order, and Karma has some kind of concept of parameterized tests, it would look cleaner to have each of these as a separate parameterized run of the test.  Otherwise, I guess it looks fine but still feels weird.

Reply:
This is kind of how I do it in all the other tests, I think it's just a side effect of how mockBackend.connections is an Observable.

Reply:
I can look into this more, but at a later date since it's a bigger conceptual change.